### PR TITLE
Mock test try

### PR DESCRIPTION
**Pull request by nitesh-chauhan**
code coverage fix for mock test.

### CODE REVIEW

File: src/main/java/com/mediamath/terminalone/TerminalOne.java [hunk 1]
```diff
@@ -128,14 +128,18 @@ private void login(String username, String password, String apiKey) throws Clien
...
     if (checkUserAndUserData() && checkUserSession() && checkUserSessionID()) {
         this.authenticated = true;
     }  
   }
+  
+  private void setServices() {
+    postService = new PostService(this.connection, this.user, this.tOneService);
+    getService = new GetService();
+    reportService = new ReportService();
+  }
```
Comment (karthik-mediamath) [inline src/main/java/com/mediamath/terminalone/TerminalOne.java]: Better approach. +1

File: src/main/java/com/mediamath/terminalone/service/PostService.java [hunk 2]
```diff
@@ -186,9 +208,9 @@ public Strategy save(Strategy entity) throws ClientException, ParseException {
 
       String path = t1Service.constructUrl(uri);
 
-      Response responseObj = this.connection.post(path, StrategyHelper.getForm(entity), this.user);
-
-      finalJsonResponse = getJsonResponse(entity, responseObj);
+      String responseString = getStrategyResponseString(entity, path);
+
```
Comment (karthik-mediamath) [inline src/main/java/com/mediamath/terminalone/service/PostService.java]: Unwanted white space.

File: src/main/java/com/mediamath/terminalone/service/PostService.java [hunk 3]
```diff
@@ -197,40 +219,29 @@ public Strategy save(Strategy entity) throws ClientException, ParseException {
...
     JsonResponse<? extends T1Entity> finalJsonResponse;
 
-    String response = responseObj.readEntity(String.class);
-
     T1JsonToObjParser parser = new T1JsonToObjParser();
 
     if (response.isEmpty())
       return null;
 
-    JsonPostErrorResponse error = jsonPostErrorResponseParser(response, responseObj);
```
Comment (karthik-mediamath) [inline src/main/java/com/mediamath/terminalone/service/PostService.java]: How is this error handled now?

File: src/main/java/com/mediamath/terminalone/service/PostService.java (hunk 2 above)
Comment (nitesh-chauhan) [inline src/main/java/com/mediamath/terminalone/service/PostService.java]: will format entire file once you merge this PR.. then you can just merge the new pr without reviewing.

File: src/main/java/com/mediamath/terminalone/service/PostService.java (hunk 3 above)
Comment (nitesh-chauhan) [inline src/main/java/com/mediamath/terminalone/service/PostService.java]: the code is refactored / moved to a different method. its in readPostResponseToString() method.  so the logic is the same error is parsed / checked before we create the finalJsonResponse entity.

**Pull request merged**